maas/resource_maas_volume_group.go

```diff
@@ -0,0 +1,259 @@
+package maas
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strconv"
+	"strings"
+
+	"github.com/canonical/gomaasclient/client"
+	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceMaasVolumeGroup() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a resource to manage MAAS Volume Groups, and construct them from partion-less block devices.",
+		CreateContext: resourceMaasVolumeGroupCreate,
+		ReadContext:   resourceMaasVolumeGroupRead,
+		UpdateContext: resourceMaasVolumeGroupUpdate,
+		DeleteContext: resourceMaasVolumeGroupDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceMaasVolumeGroupImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"available_size": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "The volume group available size (B).",
+			},
+			"block_devices": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "The list of block device ids to be included in this volume group.",
+			},
+			"machine": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The machine identifier (system ID, hostname, or FQDN) that owns the volume group.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name for this volume group",
+			},
+			"size": {
```

Comment:
Maybe we need to convert it in GB like we do for maas_block_device: `Size:      int64(d.Get("size_gigabytes").(int)) * 1024 * 1024 * 1024`. Is there any benefit of keeping it in bytes?

Reply:
Or even better, what if we consume `HumanSize` from the API response?